### PR TITLE
don't fetch a million blocks for their height

### PR DESCRIPTION
**Pull request by evanmcc**
this PR both increases the pace of poc gc (trying to make them cheaper), and also fetches only the block hash for the relevant blocks from the disk, which should make the gc dramatically cheaper.

it also removes a large sync write from the state channel client which doesn't seem to be required.

### CODE REVIEW

**Review by mrallen1 — APPROVED** [review]
lgtm - imo would be worth a couple comments in the PR how this is intended to conserve memory

File: src/ledger/v1/blockchain_ledger_v1.erl [hunk 1]
```diff
@@ -1581,11 +1581,22 @@ maybe_gc_pocs(Chain, Ledger) ->
                   {ok, V} -> V;
                   _ -> 1
               end,
-    case Version > 3 andalso Height rem 100 == 0 of
+    case Version > 3 andalso Height rem 51 == 0 of
```
Comment (macpie) [inline src/ledger/v1/blockchain_ledger_v1.erl:1584]: 👍🏻

**Review by macpie — APPROVED** [review]

File: src/ledger/v1/blockchain_ledger_v1.erl (hunk 1 above)
Comment (macpie) [reply src/ledger/v1/blockchain_ledger_v1.erl:1584]: Could this cause any sync issue?

File: src/ledger/v1/blockchain_ledger_snapshot_v1.erl [hunk 2]
```diff
@@ -464,7 +464,7 @@ import(Chain, SHA,
                                   end,
 
                               Ht = blockchain_block:height(Block),
-                              case blockchain:get_block(Ht, Chain) of
+                              case blockchain:get_block_hash(Ht, Chain) of
                                   {ok, _Block} ->
```
Comment (mrallen1) [inline src/ledger/v1/blockchain_ledger_snapshot_v1.erl:468]: `_Hash` here?

**Review by mrallen1 — APPROVED** [review]
just the one suggestion

File: src/ledger/v1/blockchain_ledger_snapshot_v1.erl (hunk 2 above)
Comment (evanmcc) [reply src/ledger/v1/blockchain_ledger_snapshot_v1.erl:468]: will do

File: src/ledger/v1/blockchain_ledger_v1.erl (hunk 1 above)
Comment (evanmcc) [reply src/ledger/v1/blockchain_ledger_v1.erl:1584]: no, it won't.  the threshold for deletion is still the same, it just happens more often

**Pull request merged**